tools/building.py

```diff
@@ -702,13 +702,19 @@ def eval_ctors(js_file, wasm_file, debug_info=False): # noqa
 
 
 def get_closure_compiler():
-  # First check if the user configured a specific CLOSURE_COMPILER in thier settings
+  # First check if the user configured a specific CLOSURE_COMPILER in their settings
   if config.CLOSURE_COMPILER:
     return config.CLOSURE_COMPILER
 
-  # Otherwise use the one installed vai npm
-  cmd = shared.get_npm_cmd('google-closure-compiler')
-  if not WINDOWS:
+  # Otherwise use the one installed via npm.
+  if WINDOWS:
+    cmd = shared.get_npm_cmd('google-closure-compiler')
+  else:
+    #  We do not want to get the one in node_modules/.bin because that's a symlink, which
+    # may not work correctly in all environments (e.g. Bazel remote execution). Instead,
+    # we go to what that symlink would be pointing to directly.
+    cmd = shared.get_npm_cmd(os.path.join('node_modules', 'google-closure-compiler', 'cli.js'),
+                             relative_to_node_bin=False)
```

Comment:
I'm still not sure I follow this. `relative_no_node_bin` is now false, so I'd expect an absolute path. But the path is `node_modules/.../...js`, which is relative to something. But relative to what?

Reply:
The emscripten root, which is where node_modules is created. Bazel doesn't work well with absolute paths, because in order for things to be hermetic, they need to be contained in the Bazel cache (also known as the external folder)

Reply:
I see, thanks. Sorry for all the questions, I'm not very familiar with this area.